packages/adblocker/src/engine/engine.ts

```diff
@@ -86,16 +86,72 @@ export interface Caching {
   write: (path: string, buffer: Uint8Array) => Promise<void>;
 }
 
-export default class FilterEngine extends EventEmitter<
-  | 'csp-injected'
-  | 'html-filtered'
-  | 'request-allowed'
-  | 'request-blocked'
-  | 'request-redirected'
-  | 'request-whitelisted'
-  | 'script-injected'
-  | 'style-injected'
-> {
+// Cosmetic rule is commonly used word in the project,
+// but cosmetic filter is used for better documentation.
+export type NetworkFilterMatchingContext = {
+  request: Request;
+
+  // `type` will cause a conflict in the variable name of `Request`.
+  matchType: FilterType.NETWORK;
```

Comment:
Nit. How about this name instead and then I am not sure if the comment is needed (it doesn't seem ambiguous):
```suggestion
  filterType: FilterType.NETWORK;
```

Reply:
Included in the updates.